src/store/modules/schedule.ts

```diff
@@ -28,9 +32,70 @@ export default class Schedule extends VuexModule {
     }
   }
 
+  get getCourseSets() {
+    return this.courseSets[this.currentTerm];
+  }
+
+  @Mutation
+  switchCurrentCourseSet(p: { name: string }): void {
+    for (const sec in this.courseSets[this.currentTerm][
+      this.currentCourseSet
+    ]) {
+      worker.setSelected(sec, false);
+    }
+    this.currentCourseSet = p.name;
+    for (const sec in this.courseSets[this.currentTerm][
+      this.currentCourseSet
+    ]) {
+      if (this.courseSets[this.currentTerm][this.currentCourseSet][sec]) {
+        worker.setSelected(sec, true);
+      }
+    }
+  }
+
+  @Mutation
+  newCourseSet(p: { name: string }): void {
+    Vue.set(this.courseSets[this.currentTerm], p.name, {});
+  }
+
+  @Action
```

Comment:
This can be a mutation I think.

Reply:
same with what I said above, you cannot call a mutation from a mutation. vuex is dumb